docs/csharp/whats-new/csharp-version-history.md

```diff
@@ -0,0 +1,124 @@
+
+The next version did introduce some cool stuff, though.
+
+- [Dynamic binding](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/dynamic)
+- [Named/optional arguments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments)
+- [Generic covariant and contravariant](https://docs.microsoft.com/en-us/dotnet/standard/generics/covariance-and-contravariance)
+- [Embedded interop types](https://stackoverflow.com/questions/20514240/whats-the-difference-setting-embed-interop-types-true-and-false-in-visual-studi)
+
+Embedded interop types alleviated a deployment pain.  Generic covariance and contravariance give you a lot of power, but they&#39;re a bit academic and probably most appreciated by framework and library authors.  Named and optional parameters let you eliminate a lot of method overloads and provide convenience.  But none of those are exactly paradigm altering.
+
+The major feature was the introduction of the dynamic keyword.  The `dynamic` introduced into C# Version 4 the ability to override the compiler on compile time typing.  By using the dynamic keyword, you can create constructs similar to dynamically typed languages like JavaScript.  You can create a dynamic x = &quot;a string&quot; and then add six to it, leaving it up to the runtime to sort out what on should happen next.
+
+This gives you the potential for errors but also great power within the language.
+
+## C# Version 5
+
+C# Version 5 wasa very focused version of the language.  Nearly all of their effort for that version into another pretty groundbreaking language concept.  Here is the major features list.
+
+- [Asynchronous members](https://msdn.microsoft.com/library/hh191443(vs.110).aspx)
+- [Caller info attributes](https://www.codeproject.com/Tips/606379/Caller-Info-Attributes-in-Csharp)
+
+The caller info attribute is pretty cool.  It lets you easily retrieve information about the context in which you&#39;re running without resorting to a ton of boilerplate reflection code.  Many developers love this feature.
+
+But async and await are the real stars of this release.  When this came out in 2012, C# changed the game again by baking asynchrony into the language as a first class participant.  If you&#39;ve ever dealt with long running operations and the implementation of webs of callbacks, you probably loved this language feature.
+
+## C# Version 6
+
+With versions 3 and 5, C# had added some pretty impressive stuff in an OO language.  (Version 2 did as well, but it was fast following Java with those language features.)  With version 6, it would go away from doing a dominant killer feature and instead release a lot of features that delighted users of the language.  Here are some of them.
```

Comment:
remove pretty

replace stuff with something less informal

a lot of features -> many features